hubconf.py

```diff
@@ -96,41 +84,49 @@ def get_module(self):
             transformer = torch.jit.script(transformer)
         batch = list(validation_data)[0]
         src_seq = patch_src(batch.src, self.opt.src_pad_idx).to(self.device)
-        trg_seq, _ = map(lambda x: x.to(self.device), patch_trg(batch.trg, self.opt.trg_pad_idx))
+        trg_seq, self.gold = map(lambda x: x.to(self.device), patch_trg(batch.trg, self.opt.trg_pad_idx))
 
         # We use validation_data for training as well so that it can finish fast enough.
         return transformer, (src_seq, trg_seq)
 
     @skipIfNotImplemented
     def eval(self, niter=1):
-        m, example_inputs = self.get_module()
-        m.eval()
+        self.module.eval()
         for _ in range(niter):
-            m(*example_inputs)
+            self.module(*self.example_inputs)
 
     @skipIfNotImplemented
     def train(self, niter=1):
-        m, _ = self.get_module()
         optimizer = ScheduledOptim(
-            optim.Adam(m.parameters(), betas=(0.9, 0.98), eps=1e-09),
+            optim.Adam(self.module.parameters(), betas=(0.9, 0.98), eps=1e-09),
             2.0, self.opt.d_model, self.opt.n_warmup_steps)
-        training_data, _ = prepare_dataloaders(self.opt, self.device)
-        batch = list(training_data)[0]
-        src_seq = patch_src(batch.src, self.opt.src_pad_idx).to(self.device)
-        trg_seq, gold = map(lambda x: x.to(self.device), patch_trg(batch.trg, self.opt.trg_pad_idx))
         for _ in range(niter):
             optimizer.zero_grad()
-            pred = m(src_seq, trg_seq)
+            pred = self.module(*self.example_inputs)
 
             loss, n_correct, n_word = cal_performance(
-                pred, gold, self.opt.trg_pad_idx, smoothing=self.opt.label_smoothing)
+                pred, self.gold, self.opt.trg_pad_idx, smoothing=self.opt.label_smoothing)
             loss.backward()
             optimizer.step_and_update_lr()
 
 
 if __name__ == '__main__':
     m = Model(device='cuda', jit=False)
-    model, example_inputs = m.get_module()
-    model(*example_inputs)
-    m.train()
-    m.eval()
+    t2 = time.time()
+    m.get_module()
+    t3 = time.time()
+
+    m.module(*m.example_inputs)
```

Comment:
This is not aligned with our API: here it's enforcing every Model to have `module/example_inputs` as attributes which is not always true, e.g GAN has two modules. 
I might prefer the old validation here?
```
    model, example_inputs = m.get_module()
    model(*example_inputs)
```

Reply:
yea, this isn't good. i did it a different way in my BERT.  What do you think about [this approach](https://github.com/wconstab/BERT-pytorch/commit/f69fffe122eeb2acfdfb0013fc1105ce52ff0881#diff-5718e9746999936f6f24ba7eabe6cc2dR68)?

Reply:
Also, i can delete the timing code.  It isn't a terrible thing to have but it makes things a bit messy, so maybe you prefer it gone?

Reply:
W.r.t I'm fine moving the slicing logic to `get_module()` so that we can keep `model(*exmaple_inputs)` here ;)  
Yea let's delete the timing code here and keep timing stuff in your bench.py in hub ;) Thanks!